### DISCUSSION TITLE
Database Integration

### DISCUSSION THREAD

**Discussion by bbardsley** [thuliteio/doks#1034] [Q&A] [answered]
I am a user of Doks, and I have been using it to record quotes from the books I'm reading. However, I have recently realized that there could be a more efficient way of doing this using databases instead of copy-pasting quotes into shortcodes.

After some research, I found that Hugo handles .yaml files quite well, and it can generate a new page for every author and/or book title, listing all the corresponding quotes. I believe this would be a great feature to have in Doks, and ChatGPT has confirmed that it should be possible to achieve this.

However, the suggested code by ChatGPT does not seem to work well with Doks at the moment. This is due to the way that  I am wondering if you have any suggestions on how to implement this feature in an easy fashion. If it is possible, I am willing to wait for the next hyas update to see if this can be done more easily.

Thank you for your hard work and dedication to improving Doks.

Best regards,  
Bernard

**Comment by h-enk** [comment]
Unfortunately, Hugo does not support building pages from data source — yet. See also: https://github.com/gohugoio/hugo/issues/5074

However, Regis' work-around should also work with Hyas/Doks: [Toward using a Headless CMS with Hugo: Building Pages from Data](https://www.thenewdynamic.com/article/toward-using-a-headless-cms-with-hugo-part-1/)

**Reply by bbardsley** [reply]
Thank you for your quick response. This might indeed work. There are however two issues I run into. 
- Whenever I'm trying to add tags (or other created taxonomies) outside of blogs, I get an error of the form

`Error: Error building site: failed to render pages: render of "term" failed: "../layouts/_default/list.html:19:13": execute of template failed: template: _default/list.html:19:13: executing "main" at <partial "main/blog-meta.html" .>: error calling partial: "../layouts/partials/main/blog-meta.html:1:17": execute of template failed: template: partials/main/blog-meta.html:1:17: executing "partials/main/blog-meta.html" at <len .Params.contributors>: error calling len: reflect: call of reflect.Value.Type on zero Value`

- How do I create a seperate custom layout outside of the default layout for a new menu. I think this might be related to the previous error (but I would like to know it outside of that). The command `layout: custom_layout`, does not seem to work. 

Thank a lot.

Kind regards,
Bernard

**Comment by h-enk** [comment, marked as answer]
- Add `./layouts/_default/terms.html` and `./layouts/_default/term.html` (based on `./layouts/_default/list.html`)
- It should work when you have a `./layouts/_default/custom_layout.html`

For more background, see [Hugo's Lookup Order](https://gohugo.io/templates/lookup-order/)

**Reply by bbardsley** [reply]
That solves it. Thanks a lot. For anyone who might read this, in https://www.thenewdynamic.com/article/toward-using-a-headless-cms-with-hugo-part-1/ there are many typo's, but when you fix them, it should work.
